# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/squashfs-tools-getnprocs.patch
# Copyright (C) 2006 - 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./squashfs-tools/mksquashfs.c.orig	2006-08-31 00:07:37.000000000 +0200
+++ ./squashfs-tools/mksquashfs.c	2006-09-01 14:20:24.000000000 +0200
@@ -2625,8 +2625,12 @@
 			processors = 1;
 		}
 #else
+#ifdef __UCLIBC__
+		processors = 1;
+#else
 		processors = get_nprocs();
 #endif
+#endif
 	}
 
 	if((thread = malloc((2 + processors * 2) * sizeof(pthread_t))) == NULL)
--- squashfs3.4/squashfs-tools/unsquashfs.c.orig	2008-08-26 09:01:54.000000000 +0200
+++ squashfs3.4/squashfs-tools/unsquashfs.c	2008-10-28 16:09:44.000000000 +0100
@@ -2428,8 +2428,13 @@
 			processors = 1;
 		}
 #else
+#ifdef __UCLIBC__
+		processors = 1;
+#else
+
 		processors = get_nprocs();
 #endif
+#endif
 	}
 
 	if((thread = malloc((3 + processors) * sizeof(pthread_t))) == NULL)
